scripts/bake-dependencies.sh

```
@@ -0,0 +1,186 @@
+# Run cmake configure to download all dependencies
+# This will download: Zig, WebKit, BoringSSL, and all other dependencies
+print "Running CMake configure to download dependencies..."
+
+# Run cmake configure - this downloads WebKit
+if [ -n "$ABI" ]; then
+    cmake -S "$BUN_REPO_PATH" -B "$BUILD_PATH" -G Ninja -DCMAKE_BUILD_TYPE=Release -DCI=ON -DABI="$ABI"
+else
+    cmake -S "$BUN_REPO_PATH" -B "$BUILD_PATH" -G Ninja -DCMAKE_BUILD_TYPE=Release -DCI=ON
+fi
+
+if [ $? -ne 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)